AUTH48 changes (#133)
quicwg/version-negotiation: draft-ietf-quic-version-negotiation.md

It is beneficial to avoid additional round trips whenever possible, especially
given that most incremental versions are broadly similar to the previous
version. This specification also defines a simple version negotiation mechanism

> that leverages similarities between versions and can negotiate between

Reviewer: This is unnecessary. 

http://itre.cis.upenn.edu/~myl/languagelog/archives/001484.html
```suggestion
which leverages similarities between versions and can negotiate between
```

# Version Negotiation Mechanism

This document specifies two means of performing version negotiation:
"incompatible", which requires a round trip and is applicable to all versions,

> and "compatible", which allows saving the round trip but only applies when the

Reviewer: ```suggestion
and "compatible", that allows saving the round trip but only applies when the
```

server MAY add reserved versions (as defined in {{Section 6.3 of QUIC}}) in
Supported Version fields.

Clients will ignore a Version Negotiation packet if it contains the Original
Version attempted by the client (see {{downgrade}}). The client also ignores a
Version Negotiation packet that contains incorrect connection ID fields (see
{{Section 6 of QUIC-INVARIANTS}}).

Upon receiving the Version Negotiation packet, the client SHALL search for a
version it supports in the list provided by the server. If it doesn't find one,
it SHALL abort the connection attempt. Otherwise, it SHALL select a mutually
supported version and send a new first flight with that version -- this version
is now the Negotiated Version.

The new first flight will allow the endpoints to establish a connection using
the Negotiated Version. The handshake of the Negotiated Version will exchange
Version Information (see {{vers-info}}) that is required to ensure that version
negotiation was genuine, i.e., that no attacker injected packets in order to
influence the version negotiation process (see {{downgrade}}).

> Only servers can start incompatible version negotiation, i.e., clients MUST NOT send

Reviewer: ```suggestion
Only servers can start incompatible version negotiation. Clients MUST NOT send
```

it into a first flight of packets from version B. As an example, if versions A
and B are absolutely equal in their wire image and behavior during the handshake
but differ after the handshake, then A is compatible with B and B is compatible

> with A. Note that the conversion of the first flight can be lossy, i.e., some data,

Reviewer: ```suggestion
with A. Note that the conversion of the first flight can be lossy; some data,
```

such as QUIC version 1 0-RTT packets, could be ignored during conversion and
retransmitted later.

> Version compatibility is not symmetric, i.e., it is possible for version A to be

Reviewer: ```suggestion
Version compatibility is not symmetric. It is possible for version A to be
```

The content of each field is described below:

Chosen Version:

> : This is the version that the sender has chosen to use for this connection. In most

Reviewer: ```suggestion
: The version that the sender has chosen to use for this connection. In most
```

cases, this field will be equal to the value of the Version field in the long
header that carries this data; however, future versions or extensions can choose
to set different values in the long header Version field.

> The content of the Available Versions field depends on whether it is sent by

Reviewer: ```suggestion
The contents of the Available Versions field depend on whether it is sent by
```

If a deployment contains multiple server instances, software updates may not
happen at exactly the same time on all server instances. Because of this, a
client might receive a Version Negotiation packet from a server instance that
has already been updated, and the client's resulting connection attempt might

> reach a different server instance that hasn't been updated yet.

Reviewer: ```suggestion
reach a different server instance which hasn't been updated yet.
```

future versions SHOULD attempt to design their new version such that commonly
deployed versions are compatible with it.

> QUIC version 1 defines multiple features that are not documented in the QUIC

Reviewer: ```suggestion
QUIC version 1 defines multiple features whch are not documented in the QUIC
```
